azurerm/resource_arm_subnet.go

```diff
@@ -66,6 +66,39 @@ func resourceArmSubnet() *schema.Resource {
 				Optional: true,
 				Elem:     &schema.Schema{Type: schema.TypeString},
 			},
+
+			"delegation": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"service_delegation": {
+							Type:     schema.TypeList,
+							Required: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"service_name": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"actions": {
+										Type:     schema.TypeList,
+										Optional: true,
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
```

Comment:
given the documentation below says there's only a single value permissible here, we could probably add validation for this:
```suggestion
											Type: schema.TypeString,
											ValidateFunc: validation.StringInSlice([]string{
											  "Microsoft.Network/virtualNetworks/subnets/action",
											}, false),
```